exporter/googlecloudstorageexporter/storage_client.go

```diff
@@ -0,0 +1,156 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlecloudstorageexporter // import "github.com/observiq/bindplane-otel-collector/exporter/googlecloudstorageexporter"
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"cloud.google.com/go/storage"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/option"
+)
+
+// storageClient is a wrapper for a Google Cloud Storage client to allow mocking for testing.
+//
+//go:generate mockery --name storageClient --output ./internal/mocks --with-expecter --filename mock_storage_client.go --structname mockStorageClient
+type storageClient interface {
+	UploadObject(ctx context.Context, objectName string, buffer []byte) error
+}
+
+// googleCloudStorageClient is the google cloud storage implementation of the storageClient
+type googleCloudStorageClient struct {
+	storageClient *storage.Client
+	config        *Config
+}
+
+// newGoogleCloudStorageClient creates a new googleCloudStorageClient with the given config
+func newGoogleCloudStorageClient(cfg *Config) (*googleCloudStorageClient, error) {
+	ctx := context.Background()
+	var opts []option.ClientOption
+
+	// Handle credentials if provided, otherwise use default credentials
+	switch {
+	case cfg.Credentials != "":
+		opts = append(opts, option.WithCredentialsJSON([]byte(cfg.Credentials)))
+		if cfg.ProjectID == "" {
+			creds, err := google.CredentialsFromJSON(ctx, []byte(cfg.Credentials))
+			if err != nil {
+				return nil, fmt.Errorf("credentials from json: %w", err)
+			}
+			cfg.ProjectID = creds.ProjectID
+		}
+	case cfg.CredentialsFile != "":
+		opts = append(opts, option.WithCredentialsFile(cfg.CredentialsFile))
+		if cfg.ProjectID == "" {
+			credBytes, err := os.ReadFile(cfg.CredentialsFile)
+			if err != nil {
+				return nil, fmt.Errorf("read credentials file: %w", err)
+			}
+			creds, err := google.CredentialsFromJSON(ctx, credBytes)
+			if err != nil {
+				return nil, fmt.Errorf("credentials from json: %w", err)
+			}
+			cfg.ProjectID = creds.ProjectID
+		}
+	default:
+		// Find application default credentials from the environment
+		creds, err := google.FindDefaultCredentials(ctx, storage.ScopeFullControl)
+		if err != nil {
+			return nil, fmt.Errorf("find default credentials: %w", err)
+		}
+		opts = append(opts, option.WithCredentials(creds))
+		if cfg.ProjectID == "" {
+			cfg.ProjectID = creds.ProjectID
+		}
```

Comment:
Default creds will get read even without setting opts here, but this follows the pattern of the GCP exporter.